### PR TITLE
fix(driver_matrix_tests.client): Better handling of testcase attributes

### PR DESCRIPTION
**Pull request by k0machi**
This fixes KeyError issues resulting from xUnit xmls missing attributes

### CODE REVIEW

**Review by roydahan — APPROVED** [review]

**Review by aleksbykov — APPROVED** [review]

**Comment by k0machi** [comment]
Found a bug with timestamp extraction, all 3 tests have divergent structure for it, so need to fix that before merging.

File: argus/client/driver_matrix_tests/client.py [hunk 1]
```diff
@@ -65,14 +114,26 @@ def get_driver_info(self, xml_name: str) -> dict[str, str]:
...
+        if (pass_count := suite_attribs.get("passed")):
+            return int(pass_count)
+        total = int(suite_attribs.get("tests", 0))
+        errors = int(suite_attribs.get("errors", 0))
+        skipped = int(suite_attribs.get("skipped", 0))
+        failures = int(suite_attribs.get("failures", 0))
+
+        return total - errors - skipped - failures
+
+    def parse_result_xml(self, xml_path: Path, test_type: str, env: dict[str, str] = None) -> RawMatrixTestResult:
```
[the remark this answers is not in the document]
Comment (k0machi) [reply argus/client/driver_matrix_tests/client.py]: Removed, it is indeed unused.

[the remark this answers is not in the document]
Comment (k0machi) [reply argus/client/driver_matrix_tests/client.py]: Indeed, fixed.

**Review by aleksbykov — APPROVED** [review]
LGTM

**Pull request merged**